File: app/api/v1/smart_escalation.py
# -*- coding: utf-8 -*-
"""
API Endpoints für SmartEscalationService.

KI-gestuetzte intelligente Aufgabenzuweisung:
- Empfehlungen für optimale Zuweisung
- Team-Auslastung anzeigen
- User-Scores debuggen/analysieren

Phase 2.3 der Feature-Roadmap (Januar 2026)
"""

from typing import Dict, List, Optional
from uuid import UUID

from app.core.types import JSONDict

from fastapi import APIRouter, Depends, HTTPException, Query, status, Request
from pydantic import BaseModel, Field
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.dependencies import get_current_active_user, get_db
from app.middleware.company_context import require_company
from app.core.safe_errors import safe_error_detail, safe_error_log
from app.db.models import User, Company
from app.services.collaboration.smart_escalation_service import (
    AssignmentFactor,
    AssignmentRecommendation,
    CandidateScore,
    FactorWeights,
    get_smart_escalation_service,
)
from app.core.rate_limiting import limiter, get_user_identifier

import structlog

logger = structlog.get_logger(__name__)

router = APIRouter(prefix="/smart-escalation", tags=["Smart Escalation"])


# ============================================================================
# Schemas
# ============================================================================


class FactorWeightsRequest(BaseModel):
    """Benutzerdefinierte Gewichtung der Faktoren."""

    expertise: float = Field(
        default=0.35,
        ge=0.0,
        le=1.0,
        description="Gewicht für Expertise (0-1)",
    )
    workload: float = Field(
        default=0.25,
        ge=0.0,
        le=1.0,
        description="Gewicht für Auslastung (0-1)",
    )
    availability: float = Field(
        default=0.25,
        ge=0.0,
        le=1.0,
        description="Gewicht für Verfügbarkeit (0-1)",
    )
    relationship: float = Field(
        default=0.15,
        ge=0.0,
        le=1.0,
        description="Gewicht für Kundenbeziehung (0-1)",
    )

    def to_factor_weights(self) -> FactorWeights:
        """Konvertiert zu FactorWeights."""
        return FactorWeights(
            expertise=self.expertise,
            workload=self.workload,
            availability=self.availability,
            relationship=self.relationship,
        )

    def validate_sum(self) -> bool:
        """Validiert dass Summe ~1.0 ist."""
        total = self.expertise + self.workload + self.availability + self.relationship
        return abs(total - 1.0) < 0.01


class CandidateScoreResponse(BaseModel):
    """Score-Details eines Kandidaten."""

    user_id: str
    user_email: str
    user_name: str

    expertise_score: float = Field(ge=0, le=100)
    workload_score: float = Field(ge=0, le=100)
    availability_score: float = Field(ge=0, le=100)
    relationship_score: float = Field(ge=0, le=100)
    total_score: float = Field(ge=0, le=100)

    expertise_details: JSONDict = Field(default_factory=dict)
    workload_details: JSONDict = Field(default_factory=dict)
    availability_details: JSONDict = Field(default_factory=dict)
    relationship_details: JSONDict = Field(default_factory=dict)

    is_available: bool = True
    unavailability_reason: Optional[str] = None

    @classmethod
    def from_domain(cls, score: CandidateScore) -> "CandidateScoreResponse":
        """Konvertiere Domain-Objekt zu Response."""
        return cls(
            user_id=str(score.user_id),
            user_email=score.user_email,
            user_name=score.user_name,
            expertise_score=round(score.expertise_score, 1),
            workload_score=round(score.workload_score, 1),
            availability_score=round(score.availability_score, 1),
            relationship_score=round(score.relationship_score, 1),
            total_score=round(score.total_score, 1),
            expertise_details=score.expertise_details,
            workload_details=score.workload_details,
            availability_details=score.availability_details,
            relationship_details=score.relationship_details,
            is_available=score.is_available,
            unavailability_reason=score.unavailability_reason,
        )


class AssignmentRecommendationResponse(BaseModel):
    """Empfehlung für Aufgabenzuweisung."""

    recommended_user_id: str
    recommended_user_name: str
    confidence: float = Field(ge=0, le=100, description="Konfidenz der Empfehlung")

    candidates: List[CandidateScoreResponse]

    factors_used: List[str]
    weights_used: Dict[str, float]

    explanation: str
    explanation_details: JSONDict

    @classmethod
    def from_domain(cls, rec: AssignmentRecommendation) -> "AssignmentRecommendationResponse":
        """Konvertiere Domain-Objekt zu Response."""
        return cls(
            recommended_user_id=str(rec.recommended_user_id),
            recommended_user_name=rec.recommended_user_name,
            confidence=round(rec.confidence, 1),
            candidates=[CandidateScoreResponse.from_domain(c) for c in rec.candidates],
            factors_used=[f.value for f in rec.factors_used],
            weights_used={
                "expertise": rec.weights_used.expertise,
                "workload": rec.weights_used.workload,
                "availability": rec.weights_used.availability,
                "relationship": rec.weights_used.relationship,
            },
            explanation=rec.explanation,
            explanation_details=rec.explanation_details,
        )


class TeamMemberWorkload(BaseModel):
    """Auslastung eines Team-Mitglieds."""

    user_id: str
    user_name: str
    open_items: int
    workload_score: float
    is_available: bool
    availability_score: float


class TeamWorkloadResponse(BaseModel):
    """Team-Auslastungsübersicht."""

    team_members: List[TeamMemberWorkload]
    total_open_items: int
    available_members: int
    total_members: int
    avg_items_per_member: float


class AssignmentRequest(BaseModel):
    """Request für Zuweisungsempfehlung."""

    document_id: Optional[str] = Field(
        default=None,
        description="Dokument-ID für Kontext",
    )
    document_type: Optional[str] = Field(
        default=None,
        description="Dokumenttyp für Expertise-Matching",
    )
    entity_id: Optional[str] = Field(
        default=None,
        description="Entity-ID für Relationship-Matching",
    )
    task_type: Optional[str] = Field(
        default=None,
        description="Aufgabentyp (validation, review, etc.)",
    )
    exclude_user_ids: Optional[List[str]] = Field(
        default=None,
        description="User-IDs die ausgeschlossen werden sollen",
    )
    weights: Optional[FactorWeightsRequest] = Field(
        default=None,
        description="Benutzerdefinierte Gewichtung",
    )
    max_candidates: int = Field(
        default=10,
        ge=1,
        le=50,
        description="Maximale Anzahl Kandidaten",
    )


# ============================================================================
# Endpoints
# ============================================================================


@router.post(
    "/recommend",
    response_model=AssignmentRecommendationResponse,
    summary="Zuweisungsempfehlung holen",
    description="Ermittelt die beste Zuweisung für eine Aufgabe basierend auf KI-Faktoren.",
)
@limiter.limit("30/minute", key_func=get_user_identifier)
async def get_assignment_recommendation(
    request: Request,
    body: AssignmentRequest,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
    company: Company = Depends(require_company),
) -> AssignmentRecommendationResponse:
    """Ermittle optimale Aufgabenzuweisung."""
    service = get_smart_escalation_service(db)

    # Gewichtung validieren falls angegeben
    weights = None
    if body.weights:
        if not body.weights.validate_sum():
            raise HTTPException(
                status_code=status.HTTP_400_BAD_REQUEST,
                detail="Gewichtungen müssen sich zu 1.0 summieren",
            )
        weights = body.weights.to_factor_weights()

    # Parse UUIDs
    document_id = UUID(body.document_id) if body.document_id else None
    entity_id = UUID(body.entity_id) if body.entity_id else None
    exclude_ids = (
        [UUID(uid) for uid in body.exclude_user_ids]
        if body.exclude_user_ids
        else None
    )

    recommendation = await service.get_assignment_recommendation(
        company_id=company.id,
        document_id=document_id,
        document_type=body.document_type,
        entity_id=entity_id,
        task_type=body.task_type,
        exclude_user_ids=exclude_ids,
        weights=weights,
        max_candidates=body.max_candidates,
    )

    if not recommendation:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Keine geeigneten Kandidaten gefunden",
        )

    logger.info(
        "smart_escalation_recommendation_api",
        company_id=str(company.id),
        user_id=str(current_user.id),
        recommended_user=str(recommendation.recommended_user_id),
        confidence=recommendation.confidence,
    )

    return AssignmentRecommendationResponse.from_domain(recommendation)


@router.get(
    "/recommend",
    response_model=AssignmentRecommendationResponse,
    summary="Zuweisungsempfehlung holen (GET)",
    description="Ermittelt die beste Zuweisung mit Query-Parametern.",
)
@limiter.limit("30/minute", key_func=get_user_identifier)
async def get_assignment_recommendation_query(
    request: Request,
    document_id: Optional[str] = Query(default=None, description="Dokument-ID"),
    document_type: Optional[str] = Query(default=None, description="Dokumenttyp"),
    entity_id: Optional[str] = Query(default=None, description="Entity-ID"),
    task_type: Optional[str] = Query(default=None, description="Aufgabentyp"),
    max_candidates: int = Query(default=10, ge=1, le=50),
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
    company: Company = Depends(require_company),
) -> AssignmentRecommendationResponse:
    """Ermittle optimale Aufgabenzuweisung via GET."""
    service = get_smart_escalation_service(db)

    # Parse UUIDs
    doc_uuid = UUID(document_id) if document_id else None
    ent_uuid = UUID(entity_id) if entity_id else None

    recommendation = await service.get_assignment_recommendation(
        company_id=company.id,
        document_id=doc_uuid,
        document_type=document_type,
        entity_id=ent_uuid,
        task_type=task_type,
        max_candidates=max_candidates,
    )

    if not recommendation:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Keine geeigneten Kandidaten gefunden",
        )

    return AssignmentRecommendationResponse.from_domain(recommendation)


@router.get(
    "/team-workload",
    response_model=TeamWorkloadResponse,
    summary="Team-Auslastung anzeigen",
    description="Zeigt Auslastungsübersicht für alle Team-Mitglieder.",
)
@limiter.limit("30/minute", key_func=get_user_identifier)
async def get_team_workload(
    request: Request,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
    company: Company = Depends(require_company),
) -> TeamWorkloadResponse:
    """Hole Team-Auslastungsübersicht."""
    service = get_smart_escalation_service(db)

    overview = await service.get_team_workload_overview(company.id)

    team_members = [
        TeamMemberWorkload(
            user_id=m["user_id"],
            user_name=m["user_name"],
            open_items=m["open_items"],
            workload_score=round(m["workload_score"], 1),
            is_available=m["is_available"],
            availability_score=round(m["availability_score"], 1),
        )
        for m in overview["team_members"]
    ]

    return TeamWorkloadResponse(
        team_members=team_members,
        total_open_items=overview["total_open_items"],
        available_members=overview["available_members"],
        total_members=overview["total_members"],
        avg_items_per_member=round(overview["avg_items_per_member"], 1),
    )


@router.get(
    "/user-scores/{user_id}",
    response_model=CandidateScoreResponse,
    summary="User-Scores anzeigen",
    description="Zeigt alle Scores eines bestimmten Users (für Debugging/Analyse).",
)
@limiter.limit("30/minute", key_func=get_user_identifier)
async def get_user_scores(
    request: Request,
    user_id: str,
    document_type: Optional[str] = Query(default=None, description="Dokumenttyp"),
    entity_id: Optional[str] = Query(default=None, description="Entity-ID"),
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_active_user),
    company: Company = Depends(require_company),
) -> CandidateScoreResponse:
    """Hole detaillierte Scores eines Users."""
    service = get_smart_escalation_service(db)

    try:
        user_uuid = UUID(user_id)
    except ValueError:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail="Ungültige User-ID",
        )

    ent_uuid = UUID(entity_id) if entity_id else None

    try:
        scores = await service.get_user_scores(
            user_id=user_uuid,
            company_id=company.id,
            document_type=document_type,
            entity_id=ent_uuid,
        )
    except ValueError as e:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=safe_error_detail(e, "User-Score-Abruf"),
        )

    return CandidateScoreResponse.from_domain(scores)


@router.get(
    "/factors",
    response_model=JSONDict,
    summary="Verfügbare Faktoren anzeigen",
    description="Listet alle verfügbaren Zuweisungsfaktoren und Standardgewichtungen auf.",
)
@limiter.limit("30/minute", key_func=get_user_identifier)
async def get_available_factors(
    request: Request,
    current_user: User = Depends(get_current_active_user),
) -> JSONDict:
    """Hole verfügbare Faktoren und Konfiguration."""
    default_weights = FactorWeights()

    return {
        "factors": [
            {
                "name": f.value,
                "description": _get_factor_description(f),
                "default_weight": getattr(default_weights, f.value),
            }
            for f in AssignmentFactor
        ],
        "default_weights": {
            "expertise": default_weights.expertise,
            "workload": default_weights.workload,
            "availability": default_weights.availability,
            "relationship": default_weights.relationship,
        },
        "score_range": {"min": 0, "max": 100},
        "thresholds": {
            "min_expertise_tasks": 3,
            "max_workload_items": 20,
            "expertise_period_days": 90,
            "relationship_period_days": 180,
        },
    }


def _get_factor_description(factor: AssignmentFactor) -> str:
    """Gibt deutsche Beschreibung für Faktor zurück."""
    descriptions = {
        AssignmentFactor.EXPERTISE: (
            "Erfahrung mit dem Dokumenttyp basierend auf bisheriger Verarbeitungshistorie"
        ),
        AssignmentFactor.WORKLOAD: (
            "Aktuelle Auslastung basierend auf offenen Validierungen und Aufgaben"
        ),
        AssignmentFactor.AVAILABILITY: (
            "Verfügbarkeit basierend auf Login-Aktivitaet und Status"
        ),
        AssignmentFactor.RELATIONSHIP: (
            "Vorherige Zusammenarbeit mit dem Kunden/Lieferanten"
        ),
    }
    return descriptions.get(factor, factor.value)

File: frontend/src/lib/api/services/smart-escalation.ts
/**
 * Smart Escalation API Service
 *
 * Kommuniziert mit den /api/v1/smart-escalation Endpoints
 * für KI-gestützte intelligente Aufgabenzuweisung
 *
 * Features:
 * - Zuweisungsempfehlungen basierend auf KI-Faktoren
 * - Team-Auslastungsübersicht
 * - User-Score Debugging/Analyse
 * - Verfügbare Faktoren und Gewichtungen
 *
 * Phase 2.3 der Feature-Roadmap (Januar 2026)
 */

import { AxiosError } from 'axios';
import { apiClient } from '../client';

// ==================== Error Classes ====================

export class SmartEscalationApiError extends Error {
  statusCode?: number;
  originalError?: unknown;

  constructor(message: string, statusCode?: number, originalError?: unknown) {
    super(message);
    this.name = 'SmartEscalationApiError';
    this.statusCode = statusCode;
    this.originalError = originalError;
  }
}

// ==================== Enums ====================

export type AssignmentFactor = 'expertise' | 'workload' | 'availability' | 'relationship';

// ==================== Frontend Types ====================

export interface FactorWeights {
  expertise: number;
  workload: number;
  availability: number;
  relationship: number;
}

export interface CandidateScore {
  userId: string;
  userEmail: string;
  userName: string;

  expertiseScore: number;
  workloadScore: number;
  availabilityScore: number;
  relationshipScore: number;
  totalScore: number;

  expertiseDetails: Record<string, unknown>;
  workloadDetails: Record<string, unknown>;
  availabilityDetails: Record<string, unknown>;
  relationshipDetails: Record<string, unknown>;

  isAvailable: boolean;
  unavailabilityReason?: string;
}

export interface AssignmentRecommendation {
  recommendedUserId: string;
  recommendedUserName: string;
  confidence: number;

  candidates: CandidateScore[];

  factorsUsed: AssignmentFactor[];
  weightsUsed: FactorWeights;

  explanation: string;
  explanationDetails: Record<string, unknown>;
}

export interface TeamMemberWorkload {
  userId: string;
  userName: string;
  openItems: number;
  workloadScore: number;
  isAvailable: boolean;
  availabilityScore: number;
}

export interface TeamWorkload {
  teamMembers: TeamMemberWorkload[];
  totalOpenItems: number;
  availableMembers: number;
  totalMembers: number;
  avgItemsPerMember: number;
}

export interface FactorInfo {
  name: string;
  description: string;
  defaultWeight: number;
}

export interface FactorsResponse {
  factors: FactorInfo[];
  defaultWeights: FactorWeights;
  scoreRange: { min: number; max: number };
  thresholds: {
    minExpertiseTasks: number;
    maxWorkloadItems: number;
    expertisePeriodDays: number;
    relationshipPeriodDays: number;
  };
}

// ==================== Request Types ====================

export interface AssignmentRequest {
  documentId?: string;
  documentType?: string;
  entityId?: string;
  taskType?: string;
  excludeUserIds?: string[];
  weights?: Partial<FactorWeights>;
  maxCandidates?: number;
}

export interface UserScoresFilter {
  documentType?: string;
  entityId?: string;
}

// ==================== Backend Types ====================

interface CandidateScoreBackend {
  user_id: string;
  user_email: string;
  user_name: string;

  expertise_score: number;
  workload_score: number;
  availability_score: number;
  relationship_score: number;
  total_score: number;

  expertise_details: Record<string, unknown>;
  workload_details: Record<string, unknown>;
  availability_details: Record<string, unknown>;
  relationship_details: Record<string, unknown>;

  is_available: boolean;
  unavailability_reason: string | null;
}

interface AssignmentRecommendationBackend {
  recommended_user_id: string;
  recommended_user_name: string;
  confidence: number;

  candidates: CandidateScoreBackend[];

  factors_used: string[];
  weights_used: {
    expertise: number;
    workload: number;
    availability: number;
    relationship: number;
  };

  explanation: string;
  explanation_details: Record<string, unknown>;
}

interface TeamMemberWorkloadBackend {
  user_id: string;
  user_name: string;
  open_items: number;
  workload_score: number;
  is_available: boolean;
  availability_score: number;
}

interface TeamWorkloadBackend {
  team_members: TeamMemberWorkloadBackend[];
  total_open_items: number;
  available_members: number;
  total_members: number;
  avg_items_per_member: number;
}

interface FactorInfoBackend {
  name: string;
  description: string;
  default_weight: number;
}

interface FactorsBackend {
  factors: FactorInfoBackend[];
  default_weights: {
    expertise: number;
    workload: number;
    availability: number;
    relationship: number;
  };
  score_range: { min: number; max: number };
  thresholds: {
    min_expertise_tasks: number;
    max_workload_items: number;
    expertise_period_days: number;
    relationship_period_days: number;
  };
}

// ==================== Transformers ====================

function transformCandidateScore(c: CandidateScoreBackend): CandidateScore {
  return {
    userId: c.user_id,
    userEmail: c.user_email,
    userName: c.user_name,
    expertiseScore: c.expertise_score,
    workloadScore: c.workload_score,
    availabilityScore: c.availability_score,
    relationshipScore: c.relationship_score,
    totalScore: c.total_score,
    expertiseDetails: c.expertise_details,
    workloadDetails: c.workload_details,
    availabilityDetails: c.availability_details,
    relationshipDetails: c.relationship_details,
    isAvailable: c.is_available,
    unavailabilityReason: c.unavailability_reason ?? undefined,
  };
}

function transformRecommendation(r: AssignmentRecommendationBackend): AssignmentRecommendation {
  return {
    recommendedUserId: r.recommended_user_id,
    recommendedUserName: r.recommended_user_name,
    confidence: r.confidence,
    candidates: r.candidates.map(transformCandidateScore),
    factorsUsed: r.factors_used as AssignmentFactor[],
    weightsUsed: r.weights_used,
    explanation: r.explanation,
    explanationDetails: r.explanation_details,
  };
}

function transformTeamMember(m: TeamMemberWorkloadBackend): TeamMemberWorkload {
  return {
    userId: m.user_id,
    userName: m.user_name,
    openItems: m.open_items,
    workloadScore: m.workload_score,
    isAvailable: m.is_available,
    availabilityScore: m.availability_score,
  };
}

function transformTeamWorkload(t: TeamWorkloadBackend): TeamWorkload {
  return {
    teamMembers: t.team_members.map(transformTeamMember),
    totalOpenItems: t.total_open_items,
    availableMembers: t.available_members,
    totalMembers: t.total_members,
    avgItemsPerMember: t.avg_items_per_member,
  };
}

function transformFactors(f: FactorsBackend): FactorsResponse {
  return {
    factors: f.factors.map((factor) => ({
      name: factor.name,
      description: factor.description,
      defaultWeight: factor.default_weight,
    })),
    defaultWeights: f.default_weights,
    scoreRange: f.score_range,
    thresholds: {
      minExpertiseTasks: f.thresholds.min_expertise_tasks,
      maxWorkloadItems: f.thresholds.max_workload_items,
      expertisePeriodDays: f.thresholds.expertise_period_days,
      relationshipPeriodDays: f.thresholds.relationship_period_days,
    },
  };
}

// ==================== Request Transformers ====================

function transformAssignmentRequest(r: AssignmentRequest): Record<string, unknown> {
  const result: Record<string, unknown> = {};

  if (r.documentId) result.document_id = r.documentId;
  if (r.documentType) result.document_type = r.documentType;
  if (r.entityId) result.entity_id = r.entityId;
  if (r.taskType) result.task_type = r.taskType;
  if (r.excludeUserIds) result.exclude_user_ids = r.excludeUserIds;
  if (r.maxCandidates) result.max_candidates = r.maxCandidates;

  if (r.weights) {
    result.weights = {
      expertise: r.weights.expertise ?? 0.35,
      workload: r.weights.workload ?? 0.25,
      availability: r.weights.availability ?? 0.25,
      relationship: r.weights.relationship ?? 0.15,
    };
  }

  return result;
}

// ==================== Error Handler ====================

function handleApiError(error: unknown, context: string): never {
  if (error instanceof AxiosError) {
    const statusCode = error.response?.status;
    const message = error.response?.data?.detail || error.message;

    if (statusCode === 404) {
      throw new SmartEscalationApiError(`${context}: Nicht gefunden`, 404, error);
    }

    if (statusCode === 400) {
      throw new SmartEscalationApiError(`${context}: ${message}`, 400, error);
    }

    throw new SmartEscalationApiError(`${context}: ${message}`, statusCode, error);
  }

  throw new SmartEscalationApiError(`${context}: Unbekannter Fehler`, undefined, error);
}

// ==================== Service ====================

export const smartEscalationService = {
  /**
   * Holt Zuweisungsempfehlung (POST)
   */
  getRecommendation: async (request: AssignmentRequest): Promise<AssignmentRecommendation> => {
    try {
      const response = await apiClient.post<AssignmentRecommendationBackend>(
        '/smart-escalation/recommend',
        transformAssignmentRequest(request)
      );
      return transformRecommendation(response.data);
    } catch (error) {
      handleApiError(error, 'Zuweisungsempfehlung laden');
    }
  },

  /**
   * Holt Zuweisungsempfehlung (GET mit Query-Params)
   */
  getRecommendationQuery: async (params: {
    documentId?: string;
    documentType?: string;
    entityId?: string;
    taskType?: string;
    maxCandidates?: number;
  }): Promise<AssignmentRecommendation> => {
    try {
      const queryParams = new URLSearchParams();
      if (params.documentId) queryParams.append('document_id', params.documentId);
      if (params.documentType) queryParams.append('document_type', params.documentType);
      if (params.entityId) queryParams.append('entity_id', params.entityId);
      if (params.taskType) queryParams.append('task_type', params.taskType);
      if (params.maxCandidates) queryParams.append('max_candidates', String(params.maxCandidates));

      const url = `/smart-escalation/recommend${queryParams.toString() ? `?${queryParams.toString()}` : ''}`;
      const response = await apiClient.get<AssignmentRecommendationBackend>(url);
      return transformRecommendation(response.data);
    } catch (error) {
      handleApiError(error, 'Zuweisungsempfehlung laden');
    }
  },

  /**
   * Holt Team-Auslastungsübersicht
   */
  getTeamWorkload: async (): Promise<TeamWorkload> => {
    try {
      const response = await apiClient.get<TeamWorkloadBackend>('/smart-escalation/team-workload');
      return transformTeamWorkload(response.data);
    } catch (error) {
      handleApiError(error, 'Team-Auslastung laden');
    }
  },

  /**
   * Holt User-Scores für Debugging/Analyse
   */
  getUserScores: async (userId: string, filter?: UserScoresFilter): Promise<CandidateScore> => {
    try {
      const queryParams = new URLSearchParams();
      if (filter?.documentType) queryParams.append('document_type', filter.documentType);
      if (filter?.entityId) queryParams.append('entity_id', filter.entityId);

      const url = `/smart-escalation/user-scores/${userId}${queryParams.toString() ? `?${queryParams.toString()}` : ''}`;
      const response = await apiClient.get<CandidateScoreBackend>(url);
      return transformCandidateScore(response.data);
    } catch (error) {
      handleApiError(error, 'User-Scores laden');
    }
  },

  /**
   * Holt verfügbare Faktoren und Konfiguration
   */
  getFactors: async (): Promise<FactorsResponse> => {
    try {
      const response = await apiClient.get<FactorsBackend>('/smart-escalation/factors');
      return transformFactors(response.data);
    } catch (error) {
      handleApiError(error, 'Faktoren laden');
    }
  },
};
